integration-tests/setup/testcluster.go

```diff
@@ -437,6 +437,7 @@ func NewTestConfig(originHost string, targetHost string) *config.Config {
 
 	conf.ProxyMaxClientConnections = 1000
 	conf.ProxyMaxStreamIds = 2048
+	conf.ProxyMaxPreparedStatementCacheSize = 5000
```

Comment:
The limit is based on size, rather than number of prepared statements: `The default calculated value is 1/256th of the heap or 10 MB, whichever is greater`.

When I set the default to 5000, my intention was to keep the PS cache memory footprint relatively small, given that the proxy usually runs on instances with limited resources. It is quite unlikely for applications to create a large number of prepared statements if they are using them correctly, so even if a user has multiple applications using the proxy I thought that this should be a reasonable value. On the other hand, the footprint of each statement in the proxy PS cache maps is small, so choosing a default value that is a bit higher should also be fine.

Reply:
So for a 16GB heap the limit would be about 60MB, I think it would be good to get an estimate of what size an average prepared statement has on our cache so we can come up with a good value for this limit instead of guessing blindly. 5000 sounds a bit too low but without any data on the size that each statement takes I'm just guessing blindly. Is there server metrics for the prepared cache size? We could use those metrics in a benchmark to get some data that would help us come up with a good limit.